Use pytest to test datalad-metalad in test_extensions-workflow

christian-monch opened the pull request:
This PR modifies the test_extensions-workflow to use `pytest` if the tested extension is `datalad-metalad`. The extension `datalad-metalad` recently switched from `nose` to `pytest` because `nose` does not work with python 3.10.

This should fix errors in `test (datalad-metalad)


### Changelog
#### 🛡 Tests
- switch test runner for `datalad-metalad` extension from `nose` to `pytest`.

bpoldrack reviewed (commented):
Looks right overall. However, the `Install dependencies` step further up has an unconditional `pip install nose` ATM. I'd say make it conditional now on `matrix.extension != 'datalad-metalad'?

bpoldrack commented:
Travis failure unrelated, but: Ping @yarikoptic - `Versioneer` upgrade didn't do the trick apparently.

christian-monch commented:
@bpoldrack: affre

> Looks right overall. However, the `Install dependencies` step further up has an unconditional `pip install nose` ATM. I'd say make it conditional now on `matrix.extension != 'datalad-metalad'`?

@bpoldrack : addressed your comment
